core/ecschema-locaters/src/SchemaXmlFileLocater.ts

```diff
@@ -120,3 +122,43 @@ export class SchemaXmlFileLocater extends SchemaFileLocater implements ISchemaLo
     return key;
   }
 }
+
+/**
+ * This locater is responsible for locating standard schema files that are released by the core-backend package.
+ * The locater needs an argument to the known backend assets directory where the core-backend package is installed.
+ * This can be accessed by the KnownLocations.nativeAssetsDir getter provided by core-backend.
+ *
+ * @note This locater is read-only and does not allow adding new schema search paths.
+ * @note This locater should be used as a fallback/last chance locater in the schema context as any user defined schema should have higher precedence over the standard schema.
+ * @beta This is a workaround due to the current lack of a full xml parser.
+ */
+export class PublishedSchemaXmlFileLocater extends SchemaXmlFileLocater implements ISchemaLocater {
+  /**
+   * Constructs a new PublishedSchemaXmlFileLocater
+   * @param knownBackendAssetsDir The assets directory where the core-backend package is installed. Can be accessed by the KnownLocations.nativeAssetsDir getter provided by core-backend.
+   */
+  public constructor(knownBackendAssetsDir: string) {
+    super();
+
+    // Pre-defined set of standard schema search paths
+    this.searchPaths.push(...globSync(path.join(knownBackendAssetsDir, "ECSchemas", "*/"), { windowsPathsNoEscape: true }).filter(fs.existsSync));
+  }
+
+  /**
+   * Overrides the addSchemaSearchPath method to prevent adding new schema search paths.
+   *
+   * @param _schemaPath - The schema path to add (ignored).
+   */
+  public override addSchemaSearchPath(_schemaPath: string): void {
+    return; // Do nothing, this is a read-only locator
```

Comment:
Can you elaborate on this? I'd rather actually add to searchPaths, or throw here.

Reply:
Okay, Rohit explained it to me. It still feels odd, but I understand the reasoning. Closing the conversation.